### ISSUE TITLE
Unified Dashboard: Register widgets in new context "Traffic"

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#4149]
## Feature Description

[As per the design doc](https://docs.google.com/document/d/1StGx76XXisE9MfwM1916D9fn7-H9BRiRgizRejTsecI/edit?pli=1#heading=h.iiduk2mmggv7), there should add the following widgets to the Traffic widget area:

  * All Traffic widget (`DashboardAllTrafficWidget`)
  * New Search Funnel widget (#4122)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by felixarntz** [comment]
@tofumatt Based on https://docs.google.com/document/d/1OnVv9P5Hbz-owVnzawoO-o37GLTFhyeIyBO6K5YZr38/edit?disco=AAAAP5ukqmA, I thought we would register the new widgets when they're ready, instead of blocking the general widget registration issues from them. Is that not what you meant? I'm thinking we could already move this forward to at least have the area appear correctly (with only the All Traffic widget initially), and then #4122 would be responsible for registering that new widget in the area later.

**Comment by tofumatt** [comment]
Indeed, I've modified the new widgets to specify that they should add their registration when ready so this is no longer blocked by the new widget. 👍🏻

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by johnPhillips** [comment]
@aaemnnosttv I think, looking again at the AC, that this might inadvertently be blocked by #4122? 

I've done the IB and PR to include the All Traffic Widget on the basis that it will be straightforward to register the Overall Page Metrics widget as part of that issue. Hope that's ok?

**Comment by felixarntz** [comment]
@johnPhillips @aaemnnosttv That should be good with only the All Traffic widget for now. #4122 is going to cover registering the new widget in here once it's ready. The focus of this issue here is to set up everything in that widget context which we already have.

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@tofumatt I'd just like to highlight two observations. 

1. With `unifiedDashboard` feature flag enabled, and when AdSense is not activated, there is a modal for `Additional Permissions Required` related to AdSense. To get by this, I had to enable AdSense on the tester plugin and connect. Is this expected at this stage? 

![image](https://user-images.githubusercontent.com/73545194/135872024-e3634bb9-bfd4-4a40-b436-3b44383df703.png)

2. With `unifiedDashboard` feature flag enabled, I do see the `All Traffic` widget when on Site Kit dashboard `wp-admin/admin.php?page=googlesitekit-dashboard` and underneath that I see the two AdSense widgets.  When I add a permaLink query `/wp-admin/admin.php?page=googlesitekit-dashboard&permaLink=test` I still see the All Traffic widget but no longer see the two AdSense widgets.  Is this also expected at this stage? Did not want to assume. 

![image](https://user-images.githubusercontent.com/73545194/135872302-6289dc13-51c0-4ca7-a195-a1be32dd87f4.png)

**Comment by felixarntz** [comment]
@wpdarren Your first point looks valid, this shouldn't be happening, so needs to be fixed. Your second point is expected, when looking at the results for a single URL (entity dashboard), the AdSense widgets should not be present, as those only work for the entire site level.

@tofumatt That AdSense widget probably needs to receive the `whenActive` treatment, which I assume it didn't have before just because it was only appearing on the AdSense module page, which didn't appear without the module active in the first place. Let's add the `whenActive` use here, with the following details: The widget should _only_ have a fallback component for when the module is active but not connected - i.e. if AdSense is not active, it doesn't show at all, and if AdSense is not fully connected, it shows the blue CTA to "complete module activation".

**Comment by aaemnnosttv** [comment]
Back to you for another pass @wpdarren 👍
